.github/workflows/integration-tests.yaml

```diff
@@ -52,24 +45,17 @@ jobs:
           sessioninfo::session_info(pkgs, include_base = TRUE)
         shell: Rscript {0}
 
-      - name: Set up license file
-        run: echo "$CONNECT_LICENSE_FILE" > /tmp/connect.lic
-
-      - name: Setup test environment
-        run: |
-          connectapi:::build_test_env(connect_license_path = "/tmp/connect.lic")
-        shell: Rscript {0}
-
-      - uses: r-lib/actions/check-r-package@v2
-
-      - name: Show testthat output
-        if: always()
-        run: find check -name 'testthat.Rout*' -exec cat '{}' \; || true
-        shell: bash
-
-      - name: Upload check results
-        if: failure()
-        uses: actions/upload-artifact@main
+      - name: Run integration tests
+        uses: nealrichardson/with-connect@dev
         with:
-          name: ${{ runner.os }}-${{ matrix.version }}-results
-          path: check
+          connect-version: ${{ matrix.version }}
+          connect-license: ${{ secrets.CONNECT_LICENSE_FILE }}
+          # TODO: rewrite tests to use CONNECT_* env vars directly
+          command: |
+            Rscript -e '
+              Sys.setenv(
+                CONNECTAPI_INTEGRATED="true",
```

Comment:
It's possible that this could go in the `env` of the action step, will try that. It's not possible to have an argument in `with` that is a YAML mapping, so `env` can't go there.